pyptex/__init__.py

```diff
@@ -229,298 +241,305 @@ def genname(self,pattern="fig{gencount}.eps"):
+        self.cachefilename = f'{self.filename}.pickle'
+        self.bibfilename = f'{self.filename}.bib'
+        self.pyptexfilename = f'{self.filename}.pyptex'
+        self.auxfilename = f'{self.filename}.aux'
+        self.latex = 'pdflatex --file-line-error --synctex=1'
         self.latexcommand = latexcommand
         self.disable_cache = False
         self.deps = {}
         self.lc = 0
-        self.argv = argv
+        self.argv = [] if argv is None else []
         self.__sympy_plot__ = None
         self.exitcode = 0
         self.generateddir()
         self.dep(__file__)
         self.compile()
-        print(texfilename+": pyptex compilation ends")
-
+        print(f'{texfilename}: pyptex compilation ends')
 
-    def run(self,S,k):
-        """An internal function for executing Python code"""
-        print("Executing Python code:\n"+S)
-        S = "\n"*k+S
-        glob = self.__globals__
+    def run(self, S, k):
+        """An internal function for executing Python code."""
+        print(f'Executing Python code:\n{S}')
+        S = '\n'*k + S
+        glob_ = self.__globals__
         doeval = False
         self.accum = []
-        try:
-            C = compile(S,self.texfilename,mode='eval')
+        with suppress(Exception):
+            C = compile(S, self.texfilename, mode='eval')
             doeval = True
-        except:
-            pass
-        if(doeval):
-            ret = eval(C,glob)
+        if doeval:
+            ret = eval(C, glob_)
             self.accum.append(ret)
         else:
-            C = compile(S,self.texfilename,mode='exec')
-            exec(C,glob)
-        print("Python result:\n"+str(self.accum))
+            C = compile(S, self.texfilename, mode='exec')
+            exec(C, glob_)
+        print(f'Python result:\n{self.accum!s}')
         return self.accum
-    def print(self,*argv):
+
+    def print(self, *argv):
         """If `pyp` is an object of type `pyptex`, `pyp.print(X)` causes `X` to be converted
         to its latex representation and substituted into the `a.pyptex` output file.
         The conversion is given by `sympy.latex(X)`, except that `None` is converted
         to the empty string.
 
-        Many values can be printed at once with the notation `pyp.print(X,Y,...)`"""
+        Many values can be printed at once with the notation `pyp.print(X, Y, ...)`."""
         self.accum.extend(argv)
 
-    def process(self,S,runner):
-        """An internal helper function for parsing the input file"""
-        SS = S.splitlines()
-        ln = numpy.cumsum(numpy.array(numpy.array(list(S),dtype='U1')=='\n',int))
-        ln = numpy.insert(ln,0,0)
-        def dowork(m):
-            if(m.start(1)>=0):
+    def process(self, S, runner):
+        """An internal helper function for parsing the input file."""
+        ln = numpy.cumsum(numpy.array(numpy.array(list(S), dtype='U1') == '\n', int))
+        ln = numpy.insert(ln, 0, 0)
```

Comment:
You delete the SS line which I think you're right, was a useless remain of a previous thing. Apart from the rightfully deleted SS line, github is getting a bit confused with the diff and marking everything on those two lines as having changed, and my eyes are watering a bit checking for changes, but maybe you can reassure me it's just whitespaces?

Reply:
Yup, it's just whitespace around the equal signs and commas.